lib/collection.js

```diff
@@ -1809,64 +2207,55 @@ Collection.prototype.aggregate = function(pipeline, options, callback) {
   }
 
   // If we have bypassDocumentValidation set
-  if (options.bypassDocumentValidation === true) {
-    command.bypassDocumentValidation = options.bypassDocumentValidation;
+  if (finalOptions.bypassDocumentValidation === true) {
+    command.bypassDocumentValidation = finalOptions.bypassDocumentValidation;
   }
 
   // Do we have a readConcern specified
   if (!ignoreReadConcern) {
-    decorateWithReadConcern(command, this, options);
+    decorateWithReadConcern(command, this, finalOptions);
   }
 
   // If we have allowDiskUse defined
-  if (options.allowDiskUse) command.allowDiskUse = options.allowDiskUse;
-  if (typeof options.maxTimeMS === 'number') command.maxTimeMS = options.maxTimeMS;
+  if (finalOptions.allowDiskUse) command.allowDiskUse = finalOptions.allowDiskUse;
```

Comment:
make a ticket for this to use `decorateCommand` (maxTimeMS, hint, explain, allowDiskUse)